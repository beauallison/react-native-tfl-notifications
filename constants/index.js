module.exports = {
  MODES: {
    TUBE: 'tube',
  },
  LINES: {
    TUBE: {
      BAKERLOO: 'bakerloo',
      CENTRAL: 'central',
      DISTRICT: 'district',
      CIRCLE: 'circle',
      HAMMERSMITH: 'hammersmith-city',
      JUBILEE: 'jubilee',
      METROPOLITAN: 'metropolitan',
      NORTHERN: 'northern',
      PICCADILLY: 'piccadilly',
      VICTORIA: 'victoria',
      WATERLOO: 'waterloo-city',
    },
  },
};

