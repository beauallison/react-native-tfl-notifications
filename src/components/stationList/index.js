import _ from 'lodash';
import React, { Component } from 'react';
import { FlatList } from 'react-native';
import { ListItem } from 'react-native-elements';
import propTypes from 'prop-types';

class StationList extends Component {
  constructor(props) {
    super(props);
    this.state = { isRefreshing: false };
  }

  componentDidMount() {
    if (!this.props.data) {
      this.refresh();
    }
  }

  async refresh() {
    this.setState({ isRefreshing: true });
    const data = await this.props.onRefresh;
    return this.setState({ data, isRefreshing: false });
  }

  render() {
    return (
      <FlatList
        onRefresh={() => this.refresh()}
        refreshing={this.state.isRefreshing}
        data={this.state.data}
        keyExtractor={item => item.ID}
        renderItem={({ item: { station, status } }) =>
          (<ListItem
            title={`${_.startCase(station)} - ${status}`}
            selectedBackgroundColor="grey"
          />)
        }
      />
    );
  }
}

StationList.propTypes = {
  data: propTypes.arrayOf(propTypes.shape({})),
  onRefresh: propTypes.shape({}).isRequired,
};

StationList.defaultProps = {
  data: undefined,
};

export default StationList;
