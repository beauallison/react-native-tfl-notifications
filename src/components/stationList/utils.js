import React from 'react';
import propTypes from 'prop-types';
import StationItem from './components/stationItem';

export const keyExtractor = item => item.ID;

export const renderItem = ({ item }) => ((<StationItem {...item} />));

renderItem.propTypes = {
  item: propTypes.shape({}).isRequired,
};
