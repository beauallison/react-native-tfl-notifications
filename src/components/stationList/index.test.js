import _ from 'lodash';
import { shallow } from 'enzyme';
import stationList from './index';
import { LINES } from '../../../constants';

const allStations = _.map(LINES.TUBE, (station, ID) => ({
  ID,
  station,
  status: 'Good service',
}));

describe('staionList', () => {
  it('should render a single station', () => {
    const stations = [allStations[0]];

    const wrapper = shallow(stationList({ stations }));
    expect(wrapper).toMatchSnapshot();
  });

  it('should render all stations', () => {
    const wrapper = shallow(stationList({ stations: allStations }));
    expect(wrapper).toMatchSnapshot();
  });
});
