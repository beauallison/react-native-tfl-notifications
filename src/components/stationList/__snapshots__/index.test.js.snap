// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stationList should render a flatList 1`] = `
<FlatList
  data={undefined}
  disableVirtualization={false}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  onRefresh={[Function]}
  refreshing={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;
